### PR TITLE
Preventing the TravelAnchor from being harvested when the player is not the owner and the mode Private or Protected.

### PR DESCRIPTION
**Pull request by JoduWei**

### CODE REVIEW

File: enderio-machines/src/main/java/crazypants/enderio/machines/machine/teleport/anchor/BlockTravelAnchor.java [hunk 1]
```diff
@@ -160,6 +160,22 @@ public void onBlockPlacedBy(@Nonnull World world, @Nonnull BlockPos pos, @Nonnul
     }
   }
 
+  @Override
+  public boolean removedByPlayer(@Nonnull IBlockState state, @Nonnull World world, @Nonnull BlockPos pos, @Nonnull EntityPlayer entityPlayer,
+      boolean willHarvest) {
+    TileEntity te = world.getTileEntity(pos);
```
Comment (HenryLoenwind) [inline enderio-machines/src/main/java/crazypants/enderio/machines/machine/teleport/anchor/BlockTravelAnchor.java]:
```
T te = getTileEntity(world, pos);
    if (te != null) {
```

Yes, other parts of this block are not yet updated.

File: enderio-machines/src/main/java/crazypants/enderio/machines/machine/teleport/anchor/BlockTravelAnchor.java [hunk 2]
```diff
@@ -160,6 +160,22 @@ public void onBlockPlacedBy(@Nonnull World world, @Nonnull BlockPos pos, @Nonnul
...
+
+    ITravelAccessable ta = (ITravelAccessable) te;
+    if (ta.getOwner().equals(UserIdent.create(entityPlayer.getGameProfile())) || (ta.getAccessMode() == ITravelAccessable.AccessMode.PUBLIC)) {
+      return super.removedByPlayer(state, world, pos, entityPlayer, willHarvest);
+    } else {
+
+      ChatUtil.sendNoSpam(entityPlayer, Lang.GUI_HARVEST_ERROR_PRIVATE.toChatServer(TextFormatting.RED + ta.getOwner().getPlayerName() + TextFormatting.WHITE));
```
Comment (HenryLoenwind) [inline enderio-machines/src/main/java/crazypants/enderio/machines/machine/teleport/anchor/BlockTravelAnchor.java]: Please use player.sendStatusMessage(..., true). ChatUtil.sendNoSpam() should no longer be used for one-line messages. (changing the existing ones is a Trello task)

Please put the colors into the lang file. §4 and §f, I think

File: enderio-machines/src/main/java/crazypants/enderio/machines/lang/Lang.java [hunk 3]
```diff
@@ -87,6 +87,7 @@
   GUI_AUTH_PROMPT(".gui.travel_accessable.prompt"),
   GUI_AUTH_PROMPT_BUTTON(".gui.travel_accessable.promp.login"),
   GUI_AUTH_ERROR_PRIVATE(".chat.travel_accessable.error.private"),
+  GUI_HARVEST_ERROR_PRIVATE(".chat.travel_harvest.error.private"),
```
Comment (HenryLoenwind) [inline enderio-machines/src/main/java/crazypants/enderio/machines/lang/Lang.java]: `.chat.travel_accessable.error.harvest`.

**Comment by JoduWei** [comment]
Ready for review

**Pull request merged**